src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/GVMDependenciesNode.cs

```diff
@@ -82,6 +83,12 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
                             return dependencies;
                         }
 
+                        if (!context.TypeSystemContext.SupportsUniversalCanon && instantiatedMethod.IsGenericDepthGreaterThan(SpecificCanonGVMDepthHeuristic_NonCanonDepth))
+                        {
+                            // Do not support too deep instantiation.
+                            return dependencies;
+                        }
+
                         dependencies.Add(context.MethodGenericDictionary(instantiatedMethod), "GVM Dependency - Dictionary");
```

Comment:
Doing this instead would be an acceptable thing to try:

```suggestion
                        if (!instantiatedMethod.IsGenericDepthGreaterThan(CanonGVMDepthHeuristic_NonCanonDepth))
                        {
                            dependencies.Add(context.MethodGenericDictionary(instantiatedMethod), "GVM Dependency - Dictionary");
                        }
```

I would set CanonGVMDepthHeuristic_NonCanonDepth to e.g. 256.

This would still fix #1124.

We should be able to build the generic dictionary at runtime and most of the time we actually can - there are some rough edges tracked in #537. Setting this number high should mitigate that bug (real world scenarios will unlikely be so deep, and even if they are, chances are decent things will still work because we can compute these most of the time).

Reply:
Can `CanonGVMDepthHeuristic_NonCanonDepth` be made configurable in csproj?

Reply:
My question would be, do we need two `CanonGVMDepthHeuristic_NonCanonDepth` ? One for Universal, and one for Specific like I do, or I otherthink things and only 2 depth parameters needed?